holistics/holistics.py

```diff
@@ -5,95 +5,91 @@
-            return 1
-        else:
-            return 0
+        except requests.exceptions.HTTPError as err:
+            print ("Fail to submit export")
+            raise err
 
-    def GetExportResults(self, job_id, _page_size = None, _page = None):
+    def get_export_results(self, job_id, _page_size = None, _page = None):
         print ("Getting export results... ", end='')
         tail_url = '/queries/get_export_results.json'
-        self.page['job_id']=job_id
+        self.page['job_id']=str(job_id)
         if _page_size is not None:
-            self.page['_page_size']=_page_size
+            self.page['_page_size']=str(_page_size)
         if _page is not None:
-            self.page['_page']=_page
-        res = self.GetURL(tail_url, self.page)
-        if res!=0:
+            self.page['_page']=str(_page)
+        try:
+            res = self.get_url(tail_url, self.page)
             res = res.json()
             while (res['status'] != 'success'):
                 if res['status'] == 'already_existed':
                     self.page['job_id'] = str(res['job_id'])
-                    break
                 if res['status'] == 'failure':
                     print ("Status: Failure")
-                    return 0
-                res = self.GetURL(tail_url, self.page).json()
+                    raise RuntimeError('Status of request is Failure')
+                res = self.get_url(tail_url, self.page).json()
                 time.sleep(1)
             print ("Success")
-            return 1
-        else:
-            return 0
+        except requests.exceptions.HTTPError as err:
+            print ("Fail to get export results")
+            raise err
         
-    def DownloadResults(self):
-        print ("Downloading results... ", end='')
-        tail_url = '/exports/download?job_id=' + self.page['job_id']
-        res = self.GetURL(tail_url)
-        if res!=0:
+    def download_results(self, path):
+        path = str(path)
+        try:
+            print ("Downloading results... ", end='')
+            tail_url = '/exports/download?job_id=' + self.page['job_id']        
+            res = self.get_url(tail_url)
             text = str(res.content, 'utf-8', errors='replace')
             data = pd.read_csv(io.StringIO(text))
-            self.data = True
-            print ("Success")
-            if self.path == None:
+            if path == None:
                 return data
             else:
                 try:
-                    data.to_csv(self.path, encoding='utf-8', index=False)
+                    data.to_csv(path, encoding='utf-8', index=False)
                     print ("Export to file successed!")
+                    return data
                 except:
-                    print ("Can't export file to " + self.path)
-        
+                    print ("Can't export file to " + path)
+        except (requests.exceptions.HTTPError,pd.errors.ParserError) as err:
```

Comment:
This exception is for read_csv

Reply:
So it should be handled on line 72 anh, because "Fail to download results" does not describe this error properly. In this case the result is downloaded but could not be parsed as CSV